.github/workflows/cmake.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow uses actions that are not certified by GitHub.
```

Comment:
This is just for my sanity:

Name: The name of the workflow

On: When the workflow occurs

Permissions: What the workflow is allowed to do

Jobs: What the workflow shall do

Build: What env is used

Steps: What is done to complete the job

Generate: Create the make files (???)

Build: Use the make files to compile the code

Test: Run the tests

Reply:
aye aye cap'n